fr/development/testing.rst

```diff
@@ -1442,6 +1442,110 @@ un message indiquant 1 passé et 4 assertions.
 Lorsque vous testez un Helper qui utilise d'autres Helpers, assurez-vous de
 créer un mock de la méthode ``loadHelpers`` de la classe View.
 
+.. _testing-events:
+
+Tester les Events
+=================
+
+Les :doc:`/core-libraries/events` sont un bon moyen pour découpler le code de
+votre application, mais parfois quand nous les testons, nous avons tendance à
+tester les événements dans les cas de test qui éxecutent ces événements. C'est
+une forme supplémentaire de couplage qui peut être évitée en utilisant
+à la place ``assertEventFired`` et ``assertEventFiredWith``.
+
+En poursuivant l'exemple sur les Orders, disons que nous avons les tables
+suivantes::
+
+    class OrdersTable extends Table
+    {
+
+        public function place($order)
+        {
+            if ($this->save($order)) {
+                // moved cart removal to CartsTable
+                $event = new Event('Model.Order.afterPlace', $this, [
+                    'order' => $order
+                ]);
+                $this->eventManager()->dispatch($event);
+                return true;
+            }
+            return false;
+        }
+    }
+
+    class CartsTable extends Table
+    {
+
+        public function implementedEvents()
+        {
+            return [
+                'Model.Order.afterPlace' => 'removeFromCart'
+            ];
+        }
+
+        public function removeFromCart(Event $event)
+        {
+            $order = $event->data('order');
+            $this->delete($order->cart_id);
+        }
+    }
+
+.. note::
+    Pour faire des asserts sur le fait que des événements sont déclenchés, vous
+    devez d'abord activer :ref:`tracking-events` sur le gestionnaire
+    d'événements pour lequel vous souhaitez faire des asserts.
+
+Pour tester le ``OrdersTable`` du dessus, nous activons le tracking dans
+``setUp`` puis faisons des asserts que l'événement a été déclenché, et faisons
+un assert que l'entity ``$order`` a été passée dans les données de l'événement::
+
```

Comment:
The original text could be updated also to remove the use of "we" in this sentence in favor of "you" like in most places in the book